pkgs/os-specific/darwin/aldente/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, undmg
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "aldente";
+  version = "1.22";
+
+  src = fetchurl {
+    name = "AlDente.dmg";
+    url = "https://github.com/davidwernhart/aldente-charge-limiter/releases/download/${finalAttrs.version}/AlDente.dmg";
+    hash = "sha256-y8S9SCDbvoKGtNqGQvJDSpbLhh7GqXUo7pLcMzuCnjo=";
+  };
+
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  nativeBuildInputs = [ undmg ];
+
+  # AlDente.dmg is not HFS formatted, default unpackPhase fails
+  # https://discourse.nixos.org/t/help-with-error-only-hfs-file-systems-are-supported-on-ventura
+  unpackCmd = ''
+    echo "File to unpack: $curSrc"
+    if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
+    mnt=$(mktemp -d -t ci-XXXXXXXXXX)
+
+    function finish {
+      echo "Detaching $mnt"
+      /usr/bin/hdiutil detach $mnt -force
+      rm -rf $mnt
+    }
+    trap finish EXIT
+
+    echo "Attaching $mnt"
+    /usr/bin/hdiutil attach -nobrowse -readonly $src -mountpoint $mnt
+
+    echo "What's in the mount dir"?
+    ls -la $mnt/
+
+    echo "Copying contents"
```

Comment:
```suggestion
```
phases are already announced and if things happen in the right phase there shouldn't be confusion